lib/frame_enum.ml

```diff
@@ -0,0 +1,67 @@
+open Core_kernel.Std
+
+module type E = sig
```

Comment:
The module type name should provide some insights on what modules of this type provide, e.g., `Enumerated` would be better than just plain `E`.

Reply:
Also, it should be documented.